### PR TITLE
Add Azure support for kfctl.sh

### PR DESCRIPTION
**Pull request by ritazh**
kfctl.sh will be going away eventually, so this is just a stop gap to support Azure until the go client kfctl is ready for other cloud providers.

@jlewi @kkasravi PTAL 

cc @sozercan @aronchick

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kubeflow/2557)

### CODE REVIEW

File: scripts/azure/util.sh [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+            echo "az group create failed."
+            exit 1
+        fi
+    else
+        echo "Resource group ${DEPLOYMENT_NAME} already exists, skip az group create."
+    fi
+    if ! az aks get-credentials --resource-group=${DEPLOYMENT_NAME} --name=${DEPLOYMENT_NAME} 2>/dev/null ;
+    then
+        echo "Kubernetes cluster not found, creating."
+        az aks create --resource-group ${DEPLOYMENT_NAME} --name=${DEPLOYMENT_NAME} --node-count 3 --node-vm-size "Standard_NC6" --kubernetes-version 1.12.5 --generate-ssh-keys -l "${AZ_LOCATION}" --service-principal ${AZ_CLIENT_ID} --client-secret "${AZ_CLIENT_SECRET}"
```
Comment (sozercan) [inline scripts/azure/util.sh]: should the sku be customizable? this is a pretty old sku

Comment (ritazh) [reply scripts/azure/util.sh]: done

**Comment by ritazh** [comment]
comment has been addressed in new commits

**Comment by jlewi** [comment]
The test failure looks like #2574

**Comment by jlewi** [comment]
I think #2555 might provide a temporary fix. You might just need to rebase on the latest master to pull that in.

**Comment by ritazh** [comment]
@jlewi thanks for the recommendation! Looks like it's still failing after the rebase.

**Comment by kkasravi** [comment]
We can port this to kfctl (golang) based on kfctl.sh or @ritazh you could do the same.
Another possibility is I could create the Azure plugin with hooks and specific azure API calls could be done by you. It looks like we would use the azure-sdk-for-go?

**Comment by ritazh** [comment]
@kkasravi I think adding azure under https://github.com/kubeflow/kubeflow/blob/master/bootstrap/pkg/client makes sense to me. we should use azure-sdk-for-go.

**Comment by ritazh** [comment]
_bump_ 
is there anything else I need to do?

**Comment by jlewi** [comment]
Most recent test failure:
http://testing-argo.kubeflow.org/workflows/kubeflow-test-infra/kubeflow-presubmit-kfctl-2557-72ba595-5954-ecd6?tab=workflow

I had to fetch the logs from stackdriver.

```
POD=kubeflow-presubmit-kfctl-2557-72ba595-5954-ecd6-625146308
gcloud --project=${PROJECT} logging read --format="table(timestamp, resource.labels.container_name, textPayload)" --freshness=24h      --order asc         "resource.type=\"k8s_container\" resource.labels.pod_name=\"${POD}\"  " > ~/tmp/${POD}.log.txt
```

The apply step failure is

```
ERROR: (gcloud.container.clusters.get-credentials) cluster kfctl-ecd6 is missing endpoint. Is it still PROVISIONING?
```

This is most likely because the cluster couldn't be created because us-east1-d was low on capacity.

So its a test flake.
/test all

**Comment by ritazh** [comment]
@jlewi can you PTAL at the failed job? This PR shouldnt impact any of the tests that run on GCP.

**Comment by jlewi** [comment]
There was some problems with the test-infra see #324

/test all

**Comment by jlewi** [comment]
latest failure looks like a flake with katib.
/test all

**Comment by jlewi** [comment]
Most recent failure is 

```
level=error msg="no prototype names matched 'spark-operator'"
```
Not sure where that would be coming from.

**Comment by jlewi** [comment]
Latest failures

```
Traceback (most recent call last):
 File "/usr/lib/python2.7/runpy.py", line 174, in _run_module_as_main
   "__main__", fname, loader, pkg_name)
 File "/usr/lib/python2.7/runpy.py", line 72, in _run_code
   exec code in run_globals
 File "/mnt/test-data-volume/kubeflow-presubmit-kfctl-2557-72ba595-6107-bdc7/src/kubeflow/kubeflow/testing/tf_job_simple_test.py", line 30, in <module>
   from py import tf_job_client
ImportError: cannot import name tf_job_client
```

This looks like #2574.

@ritazh sorry this is turning into such a marathon. We're hoping to make a big push on our test-infra in Q2/0..

Can you try rebasing on the latest master to pull in the fixes?

**Comment by jlewi** [comment]
Merging manually.

Sorry @ritazh

**Comment by ritazh** [comment]
Thanks @jlewi!

**Pull request merged**